spanner/src/copy_backup.php

```
@@ -0,0 +1,72 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/tree/master/spanner/README.md
+ */
+
+namespace Google\Cloud\Samples\Spanner;
+
+// [START spanner_copy_backup]
+use Google\Cloud\Spanner\Backup;
+use Google\Cloud\Spanner\SpannerClient;
+
+/**
+ * Copy a backup from another source backup.
+ * Example:
+ * ```
+ * copy_backup($instanceId, $backupId, $sourceBackupId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $backupId The Spanner backup ID.
+ * @param string $sourceBackupId The Spanner backup ID of the source.
+ */
+function copy_backup($instanceId, $backupId, $sourceBackupId)
+{
+    $spanner = new SpannerClient();
+    $instance = $spanner->instance($instanceId);
+
+    $expireTime = new \DateTime('+8 hours');
+    $backup = $instance->backup($sourceBackupId);
+    $operation = $backup->createCopy($backupId, $expireTime);
+
+    print('Waiting for operation to complete...' . PHP_EOL);
+
+    $operation->pollUntilComplete();
+    $newBackup = $instance->backup($backupId);
+    $newBackup->reload();
+
+    $ready = ($newBackup->state() == Backup::STATE_READY);
+
+    if ($ready) {
+        print('Backup is ready!' . PHP_EOL);
+        $info = $newBackup->info();
+        printf(
+            'Backup %s of size %d bytes was copied at %s from the source backup %s' . PHP_EOL,
+            basename($info['name']), $info['sizeBytes'], $info['createTime'], $sourceBackupId);
+    } else {
+        print('Backup is not ready!' . PHP_EOL);
```

This is not very helpful. Do we know why this would be? If it's a race condition maybe we could add a sleep, and if not we should throw an exception or print an error with the state like this:

```php
printf('Unexpected state: %s', Backup::name($newBackup->state()));
```

---

I modelled it using the `create_backup` sample. But, I agree, I'll try to find out the reason behind it and update both samples accordingly.

---

I am not yet sure why this state ever occurs, but the `create_backup` snippet of all other languages(except node) don't have an `else` block.